test/Client.t.sol

```diff
@@ -0,0 +1,334 @@
+        assertEq(clientContract.clientSPs(client, 1000), true);
+    }
+
+    function testUnmatchedSPs() public {
+        vm.prank(manager);
+        clientContract.addAllowedSPsForClient(client, allowedSPs_);
+
+        //SP == 1200
+        transferParams.operator_data =
+            hex"8282861904B0D82A5828000181E203922020F2B9A58BBC9D9856E52EAB85155C1BA298F7E8DF458BD20A3AD767E11572CA221908001A0007E9001A00503340190131861903E8D82A5828000181E203922020F2B9A58BBC9D9856E52EAB85155C1BA298F7E8DF458BD20A3AD767E11572CA221908001A0007E9001A0050334019013180";
+
+        vm.prank(client);
+        vm.expectRevert(abi.encodeWithSelector(Errors.NotAllowedSP.selector));
+        clientContract.transfer(transferParams);
+    }
+
+    function testInvalidAllocationRequest() public {
+        vm.prank(manager);
+        clientContract.addAllowedSPsForClient(client, allowedSPs_);
+
+        // AllocationRequest length is 7 instead of 6
+        transferParams.operator_data =
+            hex"8282871904B0D82A5828000181E203922020F2B9A58BBC9D9856E52EAB85155C1BA298F7E8DF458BD20A3AD767E11572CA221908001A0007E9001A00503340190131190131861903E8D82A5828000181E203922020F2B9A58BBC9D9856E52EAB85155C1BA298F7E8DF458BD20A3AD767E11572CA221908001A0007E9001A0050334019013180";
+
+        vm.prank(client);
+        vm.expectRevert(abi.encodeWithSelector(Errors.InvalidAllocationRequest.selector));
+        clientContract.transfer(transferParams);
+    }
+
+    function testInvalidOperatorDataLength() public {
+        vm.prank(manager);
+        clientContract.addAllowedSPsForClient(client, allowedSPs_);
+
+        // operator_data == [[]]
+        transferParams.operator_data = hex"8180";
+
+        vm.prank(client);
+        vm.expectRevert(abi.encodeWithSelector(Errors.InvalidOperatorData.selector));
+        clientContract.transfer(transferParams);
+    }
+
+    function testManagerCanAddAllowedSPs() public {
+        uint64[] memory allowedSPs = new uint64[](2);
+        allowedSPs[0] = 1;
+        allowedSPs[1] = 2;
+        vm.prank(manager);
+        clientContract.addAllowedSPsForClient(client, allowedSPs);
+
+        assertTrue(clientContract.clientSPs(client, allowedSPs[0]));
+        assertTrue(clientContract.clientSPs(client, allowedSPs[1]));
+
+        vm.prank(client);
+        vm.expectRevert(abi.encodeWithSelector(Errors.NotManager.selector));
+        clientContract.addAllowedSPsForClient(client, allowedSPs);
+    }
+
+    function testNotManager() public {
+        allowedSPs_.push(1200);
+        vm.expectRevert(abi.encodeWithSelector(Errors.NotManager.selector));
+        clientContract.addAllowedSPsForClient(client, allowedSPs_);
+    }
+
+    function testManagerCanSetManager() public {
+        vm.prank(manager);
+        clientContract.setManager(manager_);
+        assertEq(clientContract.manager(), manager_);
+
+        vm.expectRevert(abi.encodeWithSelector(Errors.NotManager.selector));
+        clientContract.setManager(manager_);
```

Comment:
This probably was meant to check if the old manager lost rights for `setManager`.
But `vm.prank` only works for a single next call, so here `setManager` is called by our contract directly.

Add another `vm.prank` or wrap everything in _startPrank modifier